L32-node-channel-API.md

```diff
@@ -0,0 +1,103 @@
+Node.js Channel API
+----
+* Author(s): mlumish
+* Approver: wenbozhu
+* Status: Draft
+* Implemented in: Node.js
+* Last updated: 
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/VR0C7DiG7bs
+
+## Abstract
+
+Expose the `Channel` class in the Node API, along with APIs for overriding the channel created in the `Client` constructor.
+
+## Background
+
+In the past, some users have requested the ability to explicitly create channels and share them among clients. In addition, another library needs to be able to intercept some channel functionality.
+
+
+### Related Proposals: 
+* A list of proposals this proposal builds on or supersedes.
+
+## Proposal
+
+We will add to the API a `Channel` class and two `Client` construction options.
+
+### New `Channel` Class API
+
+```ts
+enum ConnectivityState {
```

Comment:
should we add closed (half and/or full) too? is this how connectivity state defined in other languages?

Reply:
That enum was actually based on a slightly old version of the core's channel state enum. I fixed it, and now it matches the [connectivity state documentation](https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md).

Reply:
I would expect to see shutdown (full-close) and abort (fatal failure) .. but whatever core says :)